pkg/knit/cmd/podres.go

```diff
@@ -49,25 +55,54 @@ func NewPodResourcesCommand(knitOpts *KnitOptions) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			return showPodResources(cmd, opts, args)
 		},
-		Args: cobra.NoArgs,
+		Args: cobra.MaximumNArgs(1),
 	}
 	podRes.Flags().StringVarP(&opts.socketPath, "socket-path", "R", defaultSocketPath, "podresources API socket path.")
 	return podRes
 }
 
+func selectAction(apiName string) (func(cli kubeletpodresourcesv1.PodResourcesListerClient) error, error) {
+	switch apiName {
+	case apiCallList:
```

Comment:
not a big fan of switch/case  but if this excepted to remain small we can keep it

Reply:
I don't mind moving to a if/else chain here, so on it.